test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -17,74 +17,92 @@ limitations under the License.
-			},
-		}
-		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
-	})
+	defaultFlavor = utiltesting.MakeResourceFlavor("default").Obj()
+	gomega.Expect(k8sClient.Create(ctx, defaultFlavor)).To(gomega.Succeed())
 
-	ginkgo.When("Scheduling workloads on clusterQueues", func() {
-		var (
-			prodClusterQ *kueue.ClusterQueue
-			devClusterQ  *kueue.ClusterQueue
-			prodQueue    *kueue.LocalQueue
-			devQueue     *kueue.LocalQueue
-		)
+	ns = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "core-",
+		},
+	}
+	gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
 
-		ginkgo.BeforeEach(func() {
+	prodClusterQ = testing.MakeClusterQueue("prod-cq").
+		Cohort("all").
+		Resource(testing.MakeResource(corev1.ResourceCPU).
+			Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
+			Obj()).
+		Obj()
+	gomega.Expect(k8sClient.Create(ctx, prodClusterQ)).Should(gomega.Succeed())
 
-			prodClusterQ = testing.MakeClusterQueue("prod-cq").
-				Cohort("all").
-				Resource(testing.MakeResource(corev1.ResourceCPU).
-					Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
-					Obj()).
-				Obj()
-			gomega.Expect(k8sClient.Create(ctx, prodClusterQ)).Should(gomega.Succeed())
+	devClusterQ = testing.MakeClusterQueue("dev-cq").
+		Cohort("all").
+		Resource(testing.MakeResource(corev1.ResourceCPU).
+			Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
+			Obj()).
+		Obj()
+	gomega.Expect(k8sClient.Create(ctx, devClusterQ)).Should(gomega.Succeed())
 
-			devClusterQ = testing.MakeClusterQueue("dev-cq").
-				Cohort("all").
-				Resource(testing.MakeResource(corev1.ResourceCPU).
-					Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
-					Obj()).
-				Obj()
-			gomega.Expect(k8sClient.Create(ctx, devClusterQ)).Should(gomega.Succeed())
+	prodQueue = testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
+	gomega.Expect(k8sClient.Create(ctx, prodQueue)).Should(gomega.Succeed())
+
+	devQueue = testing.MakeLocalQueue("dev-queue", ns.Name).ClusterQueue(devClusterQ.Name).Obj()
+	gomega.Expect(k8sClient.Create(ctx, devQueue)).Should(gomega.Succeed())
+}
 
-			prodQueue = testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodClusterQ.Name).Obj()
-			gomega.Expect(k8sClient.Create(ctx, prodQueue)).Should(gomega.Succeed())
+func afterEach() {
+	gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+	util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, prodClusterQ, true)
+	util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, devClusterQ, true)
+	fwk.Teardown()
+}
 
-			devQueue = testing.MakeLocalQueue("dev-queue", ns.Name).ClusterQueue(devClusterQ.Name).Obj()
-			gomega.Expect(k8sClient.Create(ctx, devQueue)).Should(gomega.Succeed())
+var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
+
+	ginkgo.Context("Long PodsReady timeout", func() {
+
+		ginkgo.BeforeEach(func() {
+			beforeEachWithTimeout(time.Minute)
```

Comment:
This is not very idiomatic with ginkgo.

You can move this right under Describe and use JustBeforeEach https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach

Reply:
Thanks for pointing this out, updated.